crates/sargon/src/wrapped_radix_engine_toolkit/high_level/manifest_building/manifest_account_locker.rs

```diff
@@ -23,73 +24,62 @@ impl TransactionManifest {
-
-                    builder = builder.take_from_worktop(
-                        resource_arg,
-                        amount_arg,
-                        bucket,
-                    );
-                    builder = builder.deposit(claimant_arg, bucket);
+                    (resource_arg, amount_arg)
                 }
                 AccountLockerClaimableResource::NonFungible {
                     resource_address,
-                    ids,
+                    number_of_items: count,
                 } => {
                     let resource_arg: ScryptoResourceAddress =
                         resource_address.into();
-                    let ids_arg: Vec<ScryptoNonFungibleLocalId> =
-                        ids.iter().map(|id| id.clone().into()).collect();
-
-                    builder = builder.call_method(
-                        locker_address,
-                        ACCOUNT_LOCKER_CLAIM_NON_FUNGIBLES_IDENT,
-                        (claimant_arg, resource_arg, ids_arg.clone()),
-                    );
+                    let amount_arg: ScryptoDecimal192 = count.into();
+                    (resource_arg, amount_arg)
+                }
+            };
 
-                    let bucket = &bucket_factory.next();
+            builder = builder.call_method(
+                locker_address,
+                ACCOUNT_LOCKER_CLAIM_IDENT,
+                (claimant_arg, resource_arg, amount_arg),
+            );
 
-                    builder = builder.take_non_fungibles_from_worktop(
-                        resource_arg,
-                        ids_arg,
-                        bucket,
-                    );
+            let bucket = &bucket_factory.next();
 
-                    builder = builder.deposit(claimant_arg, bucket);
-                }
-            }
+            builder =
+                builder.take_from_worktop(resource_arg, amount_arg, bucket);
+            builder = builder.deposit(claimant_arg, bucket);
         }
 
         TransactionManifest::sargon_built(builder, claimant.network_id())
     }
 
     fn build_claimable_batch(
         claimable_resources: Vec<AccountLockerClaimableResource>,
-        size: usize,
+        max_size: u64,
     ) -> IndexSet<AccountLockerClaimableResource> {
-        let mut current_batch_size = 0;
-        claimable_resources
-            .into_iter()
-            .take_while(|claimable| {
-                current_batch_size += claimable.resource_count();
-                current_batch_size <= size
-            })
-            .collect()
+        let mut number_of_items_to_add = max_size;
+        let mut result = IndexSet::<AccountLockerClaimableResource>::new();
+
+        for claimable_resource in claimable_resources {
+            let updated_resource = claimable_resource
+                .coerce_number_of_items_at_most(number_of_items_to_add);
+            result.insert(updated_resource.clone());
+            number_of_items_to_add -= updated_resource.number_of_items();
```

Comment:
maybe a short comment `// can never be negative thanks to clamping performed in `coerce_number_of_items_at_most` ?